hibernate-core/src/main/java/org/hibernate/id/enhanced/NoopOptimizer.java

```diff
@@ -38,12 +38,6 @@ public Serializable generate(AccessCallback callback) {
 		// The lastSourceValue field is only accessed by tests,
 		// so this is not a concern.
 		IntegralDataTypeHolder value = callback.getNextValue();
```

Comment:
The rationale is that this is "what was always done" - in a word,migration.

It should be safe to change as long as you notate it in the release notes